toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -182,14 +186,17 @@ func createImageCustomizerParameters(ctx context.Context, buildDir string,
 		// While re-creating a disk image from the iso is technically possible,
 		// we are choosing to not implement it until there is a need.
 		if !ic.outputIsIso && !ic.outputIsPxe {
-			return nil, fmt.Errorf("cannot generate output format (%s) from the given input format (%s)", ic.outputImageFormat, ic.inputImageFormat)
+			return nil, NewImageCustomizerError(
+				ErrTypeImageConversion,
+				fmt.Sprintf("cannot generate output format (%s) from the given input format (%s)", ic.outputImageFormat, ic.inputImageFormat),
```

Comment:
Would it be possible to have the structured error have the error string and only the fields like `ic.outputImageFormat` have to be passed in? What would that potentially look like? Just answer here first don't implement anything yet.